### PR TITLE
New: Add rule no-namespace

### PR DESCRIPTION
**Pull request by weirdpattern**
This includes

New rule no-namespace
Corresponding test
Corresponding documentation
Update to README.md to include the new rule

### CODE REVIEW

**Comment by weirdpattern** [comment]
Hi @dannyfritz, could you please reference this PR from #5

**Comment by weirdpattern** [comment]
thinking about splitting this into two separate rules by removing the allowLegacyExternalModule, will work on this tonight...

**Comment by weirdpattern** [comment]
Ok, this is ready for review

File: docs/rules/no-namespace.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Disallows the use of `internal modules` and `namespaces`.
```
Comment (JamesHenry) [inline docs/rules/no-namespace.md]: Could we please call them "custom TypeScript modules" just to be 100% clear in differentiating them from ES2015 modules. "Internal" feels too vague here.

File: docs/rules/no-namespace.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+Internal modules (`module foo {}`) and namespaces (`namespace foo {}`) are considered outdated
```
Comment (JamesHenry) [inline docs/rules/no-namespace.md]: Again "custom TypeScript modules"

File: docs/rules/no-namespace.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Disallows the use of `internal modules` and `namespaces`.
+
+Internal modules (`module foo {}`) and namespaces (`namespace foo {}`) are considered outdated 
+ways to organize TypeScript code. ES2015 module syntax is now preferred.   
+
+This rule still allows the use of external modules (`declare module 'foo' {}`).
```
Comment (JamesHenry) [inline docs/rules/no-namespace.md]: the use of "TypeScript module declarations to describe external APIs"

File: docs/rules/no-namespace.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## Rule Details
+
+This rule aims to standardise the way TypeScript modules are declared.
+
+## Options
```
Comment (JamesHenry) [inline docs/rules/no-namespace.md]: It does not feel right to have a rule called "no-" and then have the exceptions enabled by default.

Please change the defaults for both exceptions to `false`.

File: lib/rules/no-namespace.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+         * @param {ASTNode} node the node to be evaluated.
+         * @returns {boolean} true when node is an external declaration.
+         * @private
+         */
+        function isExternalModule(node) {
+            return node.name && node.name.type === "Literal";
+        }
+
+        /**
+         * Determines if node is either a declaration.
```
Comment (JamesHenry) [inline lib/rules/no-namespace.js]: "either a declaration"....or?

File: tests/lib/rules/no-namespace.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
+/**
+ * @fileoverview Disallows the use of internal modules and namespaces.
```
Comment (JamesHenry) [inline tests/lib/rules/no-namespace.js]: Again please could we amend the use of "internal vs external" in this file to something more TypeScript specific. I do not feel it is clear enough what internal and external denote.

**Comment by weirdpattern** [comment]
will work on these later today!
thanks for the review

**Comment by weirdpattern** [comment]
@JamesHenry this is ready, thanks!

**Comment by JamesHenry** [comment]
Thanks a lot!

**Pull request merged**